code/game/objects/structures/cage.dm

```diff
@@ -0,0 +1,180 @@
+//Cages are structures that hold mobs
+//If its cover is opened, the mobs can interact with their surroundings (but can't move or escape)
+//If its cover is closed, the mobs inside can attempt to open it from the inside. Opening the cover takes 30 seconds
+
+//
+
+#define C_OPENED 0
+#define C_CLOSED 1
+
+/obj/structure/cage
+	name = "cage"
+	desc = "A large and heavy plasteel box, used to store dangerous animals and humans. It has two doors - the outer \"cover\", and the inner \"bars\". The cover is a thin plasteel sheet with tiny holes in the corners to let air through. The bars consist of thick plasteel rods, evenly spaced apart."
+
+	density = 1
+	anchored = 0
+
+	icon = 'icons/obj/cage.dmi'
+	icon_state = "cage_base"
+
+	var/cover_state = C_OPENED
+	var/door_state = C_CLOSED
+
+	var/damage_threshold_to_break = 100
+
+/obj/structure/cage/New()
+	..()
+
+	update_icon()
+
+/obj/structure/cage/update_icon()
+	overlays = list()
+
+	if(cover_state == C_CLOSED)
+		var/image/cover_overlay = image('icons/obj/cage.dmi', icon_state = "cage_cover", layer = OBJ_LAYER)
+		overlays += cover_overlay
+	else if(door_state == C_CLOSED) //Door is only visible when the cover is open
+		var/image/door_overlay = image('icons/obj/cage.dmi', icon_state = "cage_door", layer = MOB_LAYER + 0.5) //Above mobs
+		overlays += door_overlay
+
+/obj/structure/cage/attack_animal(mob/living/simple_animal/user)
+	if(!istype(user)) return
+
+	var/damage = rand(user.melee_damage_lower, user.melee_damage_upper)
+	if((damage >= damage_threshold_to_break))
+		if(prob(80))
+			visible_message("<span class='warning'>\The [src] shakes violently!</span>")
+		else
+			toggle_door(user)
+
+/obj/structure/cage/examine(mob/user)
+	..()
+
+	to_chat(user, "<span class='info'>Alt + click opens/closes the cage's cover.</span>")
```

Comment:
how else do you let people know how to use a new feature (considering nobody reads the changelog)